internal/services/postgres/postgresql_flexible_server_virtual_endpoint_resource.go

```diff
@@ -159,9 +160,19 @@ func (r PostgresqlFlexibleServerVirtualEndpointResource) Read() sdk.ResourceFunc
 						return metadata.MarkAsGone(id)
 					}
 
-					// Model.Properties.Members should be a tuple => [source_server, replication_server]
-					state.SourceServerId = servers.NewFlexibleServerID(id.SubscriptionId, id.ResourceGroupName, (*resp.Model.Properties.Members)[0]).ID()
-					state.ReplicaServerId = servers.NewFlexibleServerID(id.SubscriptionId, id.ResourceGroupName, (*resp.Model.Properties.Members)[1]).ID()
+					// Model.Properties.Members is a tuple => [source_server_id, replication_server_name]
+					sourceServerName := (*resp.Model.Properties.Members)[0]
+					replicaServerName := (*resp.Model.Properties.Members)[1]
+
+					sourceServerId := servers.NewFlexibleServerID(id.SubscriptionId, id.ResourceGroupName, sourceServerName).ID()
+
+					replicaServer, err := lookupFlexibleServerByName(ctx, flexibleServerClient, id, replicaServerName, sourceServerId)
+					if err != nil {
+						return err
+					}
+
+					state.SourceServerId = sourceServerId
+					state.ReplicaServerId = *replicaServer.Id
```

Comment:
Since this is direct from the lookup API, we can't be sure it will be correctly cased for the expected state (since we validate the input), can we normalise it using `servers.ParseFlexibleServerID()` before setting the value into `state.ReplicaServerId`?

Reply:
I've updated the code to reflect your suggestion